src/main/java/org/folio/helper/PurchaseOrderLineHelper.java

```diff
@@ -896,6 +899,51 @@ private CompletableFuture<Void> processPoLineEncumbrances(CompositePurchaseOrder
     return completedFuture(null);
   }
 
+  private CompletableFuture<Void> updateInventoryItemStatus(CompositePoLine compOrderLine, JsonObject lineFromStorage, RequestContext requestContext) {
+    PoLine poLineFromStorage =  lineFromStorage.mapTo(PoLine.class);
+    if (isStatusChanged(compOrderLine, poLineFromStorage) && isCurrentStatusCanceled(compOrderLine)) {
+      return purchaseOrderLineService.getPoLinesByOrderId(compOrderLine.getPurchaseOrderId(), requestContext)
+        .thenCompose(poLines -> {
+          if (poLines.size() > 1) {
+            List<String> poLineIds = poLines.stream().map(PoLine::getId).collect(toList());
+            return inventoryManager.getItemsByPoLineIdsAndStatus(poLineIds, ItemStatus.ON_ORDER.value(), requestContext)
+              .thenCompose(items -> {
+                if (items.size() > 1) {
+                  //Each poLine can have only one linked item
+                  Optional<JsonObject> poLineItem = items.stream()
+                    .filter(item -> compOrderLine.getId().equals(item.getString("purchaseOrderLineIdentifier"))).findFirst();
```

Comment:
it would be nice if `purchaseOrderLineIdentifier` is referenced as a constant somewhere. If the property is created in mod-orders, the logic that is populating this property and this reading from it, should reference that constant.